x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings.tsx

```diff
@@ -0,0 +1,139 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import {
+  EuiButton,
+  EuiConfirmModal,
+  EuiEmptyPrompt,
+  EuiOverlayMask,
+  EuiPageContent,
+  EuiPageContentBody,
+  EuiPageHeader,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { FlashMessages } from '../../../shared/flash_messages';
+import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { Loading } from '../../../shared/loading';
+import { AddRoleMappingButton, RoleMappingsTable } from '../../../shared/role_mapping';
+import {
+  EMPTY_ROLE_MAPPINGS_TITLE,
+  ROLE_MAPPINGS_TITLE,
+  ROLE_MAPPINGS_DESCRIPTION,
+} from '../../../shared/role_mapping/constants';
+
+import { ROLE_MAPPING_NEW_PATH } from '../../routes';
+
+import {
+  ROLE_MAPPINGS_ENGINE_ACCESS_HEADING,
+  EMPTY_ROLE_MAPPINGS_BODY,
+  ROLE_MAPPINGS_RESET_BUTTON,
+  ROLE_MAPPINGS_RESET_CONFIRM_TITLE,
+  ROLE_MAPPINGS_RESET_CONFIRM_BUTTON,
+  ROLE_MAPPINGS_RESET_CANCEL_BUTTON,
+} from './constants';
+import { RoleMappingsLogic } from './role_mappings_logic';
+import { generateRoleMappingPath } from './utils';
+
+export const RoleMappings: React.FC = () => {
+  const { initializeRoleMappings, handleResetMappings, resetState } = useActions(RoleMappingsLogic);
+  const { roleMappings, multipleAuthProvidersConfig, dataLoading } = useValues(RoleMappingsLogic);
+
+  useEffect(() => {
+    initializeRoleMappings();
+    return resetState;
+  }, []);
+
+  const [isResetWarningVisible, setResetWarningVisibility] = useState(false);
+  const showWarning = () => setResetWarningVisibility(true);
+  const hideWarning = () => setResetWarningVisibility(false);
+
+  if (dataLoading) return <Loading />;
+
+  const RESET_MAPPINGS_WARNING_MODAL_BODY = (
+    <FormattedMessage
+      id="xpack.enterpriseSearch.appSearch.resetMappingsWarningModalBody"
+      defaultMessage="{strongText}, and all users who successfully authenticate will be assigned the Owner role and have access to all engines."
+      values={{
+        strongText: (
+          <strong>
+            {i18n.translate('xpack.enterpriseSearch.appSearch.resetMappingsWarningModalBodyBold', {
+              defaultMessage: 'All role mappings will be deleted',
+            })}
+          </strong>
+        ),
+      }}
+    />
+  );
+
+  const addMappingButton = <AddRoleMappingButton path={ROLE_MAPPING_NEW_PATH} />;
+
+  const roleMappingEmptyState = (
+    <EuiEmptyPrompt
+      iconType="usersRolesApp"
+      title={<h2>{EMPTY_ROLE_MAPPINGS_TITLE}</h2>}
+      body={<p>{EMPTY_ROLE_MAPPINGS_BODY}</p>}
+      actions={addMappingButton}
+    />
+  );
+
+  const roleMappingsTable = (
+    <RoleMappingsTable
```

Comment:
[possibly out of scope for this PR, can be addressed in a separate PR] I just tested this table on mobile and it looks like it's hiding the per-card table headers - any idea why? IMO that's still useful information to have on mobile and I think we should keep it

<img width="765" alt="desktop" src="https://user-images.githubusercontent.com/549407/111545845-f34e2a00-8733-11eb-94f6-3b3920f1f764.png">

<img width="419" alt="mobile" src="https://user-images.githubusercontent.com/549407/111545847-f47f5700-8733-11eb-8027-83cf9ed54f01.png">

Reply:
Since this PR's goal is parity and the feature will probably change, as discussed in previous threads, I'm going to defer this for another PR.